client/blocks/editor-checkout-modal/index.tsx

```diff
@@ -23,74 +28,72 @@ import './style.scss';
-	products: Array< {
-		product_id: number;
-		product_slug: string;
-	} >;
+async function getCart( site: SiteData, cartData: CartData ) {
+	try {
+		return await wpcom.setCart( site.ID, cartData );
+	} catch {
+		return;
+	}
 }
 
-type Props = {
-	site: Site;
-	cartData: CartData;
-	onClose: () => void;
-	isOpen: boolean;
-};
-
-class EditorCheckoutModal extends Component< Props > {
-	static defaultProps = {
-		isOpen: false,
-		onClose: () => null,
-		cartData: {},
-	};
-
-	async getCart() {
-		// Important: If getCart or cartData is empty, it will redirect to the plans page in customer home.
-		const { site, cartData } = this.props;
+function fetchStripeConfigurationWpcom( args: Record< string, unknown > ) {
+	return fetchStripeConfiguration( args, wpcom );
+}
 
-		try {
-			return await wpcom.setCart( site.ID, cartData );
-		} catch {
-			return;
-		}
-	}
+const EditorCheckoutModal = ( props: Props ) => {
+	const { site, isOpen, onClose, cartData } = props;
+	const hasEmptyCart = ! cartData.products || cartData.products.length < 1;
 
-	render() {
-		const { site, isOpen, onClose, cartData } = this.props;
+	const user = userFactory();
+	const isLoggedOutCart = ! user?.get();
+	const waitForOtherCartUpdates = false;
+	// We can assume if they're accessing the checkout in an editor that they have a site.
+	const isNoSiteCart = false;
 
-		const hasEmptyCart = ! cartData.products || cartData.products.length < 1;
+	const cartKey = useMemo(
+		() =>
+			getCartKey( {
+				selectedSite: site,
+				isLoggedOutCart,
+				isNoSiteCart,
+				waitForOtherCartUpdates,
+			} ),
+		[ waitForOtherCartUpdates, site, isLoggedOutCart, isNoSiteCart ]
+	);
 
-		return hasEmptyCart ? null : (
-			<div className={ classnames( 'editor-checkout-modal', isOpen ? 'is-open' : '' ) }>
-				<div className="editor-checkout-modal__header">
-					<div className="editor-checkout-modal__wp-logo">
-						<Icon icon={ wordpress } size={ 36 } />
-					</div>
-					<Button isLink className="editor-checkout-modal__close-button" onClick={ onClose }>
-						<Icon icon={ close } size={ 24 } />
-					</Button>
+	return hasEmptyCart ? null : (
+		<div className={ classnames( 'editor-checkout-modal', isOpen ? 'is-open' : '' ) }>
+			<div className="editor-checkout-modal__header">
+				<div className="editor-checkout-modal__wp-logo">
+					<Icon icon={ wordpress } size={ 36 } />
 				</div>
+				<Button isLink className="editor-checkout-modal__close-button" onClick={ onClose }>
+					<Icon icon={ close } size={ 24 } />
+				</Button>
+			</div>
+			<ShoppingCartProvider
+				cartKey={ cartKey }
+				getCart={ () => getCart( site, cartData ) }
+				setCart={ () => getCart( site, cartData ) }
+			>
 				<StripeHookProvider fetchStripeConfiguration={ fetchStripeConfigurationWpcom }>
 					<CompositeCheckout
 						isInEditor
 						siteId={ site.ID }
 						siteSlug={ site.slug }
-						getCart={ this.getCart.bind( this ) }
+						getCart={ () => getCart( site, cartData ) }
```

Comment:
You can remove this prop entirely as it's been moved to the ShoppingCartProvider